core/src/main/java/com/sequenceiq/cloudbreak/reactor/handler/cluster/StopStartDownscaleGetRunningInstancesHandler.java

```diff
@@ -0,0 +1,127 @@
+import com.sequenceiq.cloudbreak.domain.stack.Stack;
+import com.sequenceiq.cloudbreak.domain.stack.instance.InstanceMetaData;
+import com.sequenceiq.cloudbreak.cloud.event.instance.StopStartDownscaleGetRunningInstancesRequest;
+import com.sequenceiq.cloudbreak.cloud.event.instance.StopStartDownscaleGetRunningInstancesResult;
+import com.sequenceiq.cloudbreak.service.cluster.ClusterApiConnectors;
+import com.sequenceiq.cloudbreak.service.stack.RuntimeVersionService;
+import com.sequenceiq.cloudbreak.service.stack.StackService;
+
+import reactor.bus.Event;
+import reactor.bus.EventBus;
+
+@Component
+public class StopStartDownscaleGetRunningInstancesHandler implements CloudPlatformEventHandler<StopStartDownscaleGetRunningInstancesRequest> {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(StopStartDownscaleGetRunningInstancesHandler.class);
+
+    @Inject
+    private CloudPlatformConnectors cloudPlatformConnectors;
+
+    @Inject
+    private ClusterApiConnectors clusterApiConnectors;
+
+    @Inject
+    private StackService stackService;
+
+    @Inject
+    private RuntimeVersionService runtimeVersionService;
+
+    @Inject
+    private CloudInstanceIdToInstanceMetaDataConverter cloudInstanceIdToInstanceMetaDataConverter;
+
+    @Inject
+    private InstanceMetaDataToCloudInstanceConverter instanceMetaDataToCloudInstanceConverter;
+
+    @Inject
+    private EventBus eventBus;
+
+    @Override
+    public Class<StopStartDownscaleGetRunningInstancesRequest> type() {
+        return StopStartDownscaleGetRunningInstancesRequest.class;
+    }
+
+    @Override
+    public void accept(Event<StopStartDownscaleGetRunningInstancesRequest> event) {
+        StopStartDownscaleGetRunningInstancesRequest request = event.getData();
+        LOGGER.info("StopStartDownscaleGetRunningInstancesHandler: {}", event.getData().getResourceId());
+
+        CloudContext cloudContext = request.getCloudContext();
+
+        try {
+            CloudConnector<?> connector  = cloudPlatformConnectors.get(cloudContext.getPlatformVariant());
+            AuthenticatedContext ac = getAuthenticatedContext(request, cloudContext, connector);
+
+            Stack stack = stackService.getByIdWithLists(request.getResourceId());
+            List<InstanceMetaData> startedInstancesMetadata = cloudInstanceIdToInstanceMetaDataConverter.getNotDeletedAndNotZombieInstances(
+                    stack, request.getHostGroupName(), collectStartedInstancesFromCloudProvider(connector, ac, request.getAllInstancesInHostGroup()).stream()
+                            .map(CloudInstance::getInstanceId)
+                            .collect(Collectors.toSet()));
+
+            ClusterStatusService clusterStatusService = clusterApiConnectors.getConnector(stack).clusterStatusService();
+
+            ExtendedHostStatuses extendedHostStatuses = clusterStatusService.getExtendedHostStatuses(
+                    runtimeVersionService.getRuntimeVersion(stack.getCluster().getId()));
```

Comment:
We will likely need to customize the healtchecks here. The current healthchecks consider nodes in 'maintenance mode' to be 'UNHEALTHY'. It is very possible for a node to be in 'maintenance mode' - but the NMs are still running (e.g. during the previous downscale - nodes were put into maintenance, but the decommission step failed. Note: this would happen when we change when maintenance happens - but generally safe to assume that this change will be made soon enough). Such nodes can be part of YARN, running tasks and should not be decommission candidates. Instead, they should be recommission candidates and handled during an upscale.
The healthcheck also considers a 'CONCERNING' state as HEALTHY. This is not the case, and we can make no assumptions about nodes in the 'CONCERNING' state.
At some point in the future - we may want to consider the health of the services running on a node.

Some scenarios (summarized)
- Node in maintenance mode, but services running.
- Node with CONCERNING health (we know nothing else about why this is the case)
- Services not RUNNING but the node is considered HEALTHY (Likely a result of the pre 7.2.12 API unless we can find additional info in the API result to make a sensible decision)
- All UNHEALTHY nodes are not candidates for STOP. e.g. missing nodes. Such scenarios will cause a failure of the decommission.

Generally: getting the raw healthcheck results, and processing them, or enhancing the abstraction that exists to include aspects like maintenance mode (and not collapsing them into a general node health) is required.